packages/editor/src/bindings/post-meta.js

```diff
@@ -76,20 +77,78 @@ export default {
 
 		return true;
 	},
-	getFieldsList( { registry, context } ) {
-		const metaFields = registry
-			.select( coreDataStore )
-			.getEditedEntityRecord(
+	getFieldsList: function GetFieldsList( { registry, context } ) {
+		let metaFields = {};
+		const {
+			type,
+			is_custom: isCustom,
+			slug,
+		} = registry.select( editorStore ).getCurrentPost();
+		const { getPostTypes, getEditedEntityRecord } =
+			registry.select( coreDataStore );
+
+		let postType = context?.postType;
+
+		// useSelect prevents needing a blockBindingsPanel render to fetch the data.
+		const fields = useSelect(
+			( select ) => {
+				const entityRecord = select( coreDataStore ).getEntityRecord(
+					'root',
+					'postType',
+					postType
+				);
+				return entityRecord?.meta;
+			},
+			[ postType ]
+		);
```

Comment:
I copied this code from the footnotes:
https://github.com/WordPress/gutenberg/blob/9add0faa3bd549664f560118d88ce785e6bb3230/packages/block-library/src/footnotes/format.js#L57

Ping @Mamaduka or @tyxla 

Is this correct? If we call getEntityRecord inside the conditional without the React Hook, for the first render of the BlockBindingsPanel (the one that needs the fields), `hasFinishedResolution` is still false, as is dispatched on that click.

Reply:
Seems correct to me because the network request is still resolving. Also it's expected that `hasFinishedResolution` is false and `getEntityRecord()` will return `undefined` in that case.

Reply:
Just to confirm: If `getFieldsList` is using `useSelect`, and `getFieldsList` is used inside a loop and inside a conditional, wouldn't we break [React rules of hooks](https://react.dev/reference/rules/rules-of-hooks)?

Reply:
Yes, but there are ways around it. You can have the loop inside the `useSelect` if necessary. 

Also, as a backup plan, you can always use lower-level `registry.select()` (in the example above we're already using `useRegistry`), although I'd recommend using `useSelect` when possible.

Reply:
We moved the code to an `useSelect`, now is outside a loop.

Reply:
Nice, thanks 🙌